modules/netflow/configuration/logstash/netflow.conf.erb

```diff
@@ -182,6 +182,19 @@ filter {
         }
 
         # Populate bytes transferred in the flow.
+
+        if [netflow][fwd_flow_delta_bytes] {
```

Comment:
In order to preserve behaviour when flow includes these values _and_ one of the other values that could write `[netflow][bytes]`, these should be added at the _tail_ of the `else if` chain.

We should only take one of these paths _after_ observing the absence of `[netflow][in_bytes]`, `[netflow][out_bytes]`, and `[netflow][in_permanent_bytes]`.

Reply:
IIRC, during Support Summit I handled a ticket where the user had extracted this pipeline and encountered this issue exactly (it may have been Todd's ticket). The Cisco ASA pcap showed that `fwd_flow_delta_bytes` and `rev_flow_delta_bytes` were both present and not necessary equal. Same with `initiatorPackets`, `responderPackets`, `initiatorOctets` and `responderOctets`

From a [Cisco Community Support ticket](https://community.cisco.com/t5/other-network-architecture/asa-5520-netflow-packets-don-t-contain-byte-data/td-p/2495574)
> New fields 231 (initiatorOctets) and 232 (responderOctets) will replace field 85 (IN_PERMANENT_BYTES) along with real-time flow update support in 8.4(5) and later software. However, it may take a bit for third-party Netflow Collectors to pick up these new fields as they come from IPFIX rather than legacy Netflow V9 world.

See what Robert Cowart did [here](https://github.com/robcowart/elastiflow/blob/master/logstash/elastiflow/conf.d/20_filter_30_ipfix.logstash.conf#L276)